File: selectionSort/selectionSort.js
// 选择排序算法
// Selection Sort Algorithm

function selectionSort(arr) {
    var len = arr.length;

    for ( var i = 0; i < len; i++ ) {
        var tar = i;
        for ( var j = i + 1; j < len; j++) {
            if ( arr[tar] > arr[j] ) {
                tar = j;
            }
        }
        var temp = arr[tar];
        arr[tar] = arr[i];
        arr[i] = temp;
    }
    return arr;
}

var lst = [43, 31, 43, 89, 76, 5, 65, 4, 23, 75, 90]

sorted_lst = selectionSort(lst);
console.log(sorted_lst);

File: selectionSort/selectionSort.py
from time import clock

# Selection Sort Algorithm

def selectionSort(arr):
    for i in range(0, len(arr)):
        tar = i
        for j in range(i+1, len(arr)):
            if arr[tar] > arr[j]:
                tar = j
        arr[i], arr[tar] = arr[tar], arr[i]
    return arr


if __name__ == "__main__":
    lst = [34, 56, 87, 54, 90, 43, 21, 32, 89]
    start = clock()
    arr = selectionSort(lst)
    print(arr)
    end = clock()
    print((end - start) / 1000)
